plugins/secrets/vault/opts.go

```diff
@@ -24,3 +28,21 @@ func WithRenewal(d time.Duration) Opts {
 		v.renew = d
 	}
 }
+
+func WithKubernetesAuth() Opts {
+	return func(v *vault) {
+		addr := os.Getenv("VAULT_ADDR")
+		role := os.Getenv("DRONE_VAULT_KUBERNETES_ROLE")
+		mount := os.Getenv("DRONE_VAULT_AUTH_MOUNT_POINT")
+		jwtFile := "/var/run/secrets/kubernetes.io/serviceaccount/token"
+		token, ttl, err := getKubernetesToken(addr, role, mount, jwtFile)
+		if err != nil {
+			logrus.Debugf("vault: failed to obtain token via kubernetes-auth backend: %s", err)
+			return
+		}
+
+		v.client.SetToken(token)
+		v.ttl = ttl
+		v.renew = ttl / 2
+	}
```

Comment:
typically this logic would not happen inside the option, the option would set variables in the `vault` struct and then we would do something like this:

```
// New returns a new store with secrets loaded from vault.
func New(store model.ConfigStore, opts ...Opts) (secrets.Plugin, error) {
	client, err := api.NewClient(nil)
	if err != nil {
		return nil, err
	}
	v := &vault{
		store:  store,
		client: client,
	}
	for _, opt := range opts {
		opt(v)
	}
+	if v.auth == 'kubernetes' {
+		err = v.initKubernetes()
+		if err != nil {
+			return nil, err
+		}
+	} 
	v.start() // start the refresh process.
	return v, nil
}
```

note that we could store the kube configuration in vault like this:

```
type vault struct {
	store  model.ConfigStore
	client *api.Client
	ttl    time.Duration
	renew  time.Duration
+	auth string
+	kubeAuth
+	githubAuth // for example
+
	done   chan struct{}
}

+type kubeAuth struct {
+  addr, role, mount string
+}
```